### ISSUE TITLE
The compatibility of different browsers should be considered about test case css(--customProperty)

### ISSUE THREAD

**Issue by fecore1** [jquery/jquery#4926]
### Description ###
The compatibility of different browsers should be considered about test case [css(--customProperty)](https://github.com/jquery/jquery/blob/main/test/unit/css.js#L1780)

`$elem.css( "--prop2" )` has different val on different browsers:

- firefox 91: `$elem.css( "--prop2" ) => 'val2'`
- chrome 93: `$elem.css( "--prop2" ) => ' val2'`
- edge 93: `$elem.css( "--prop2" ) => ' val2'`

As i know, `jQuery.css` is based on `getComputedStyle` and `getPropertyValue`, and that have different behaves on different browsers.

### Link to test case ###
Please open test case bellow use different browsers and attention the value of `--prop2`

https://codepen.io/fecore1/pen/KKqNxQx

**Comment by mgol** [comment]
Thanks for the report. There was a recent spec change that require CSS Custom Properties values to be trimmed and it's just been implemented in Firefox 91 so far. I expect it to take some time to roll out to all supported browsers, especially Safari so maybe we should, indeed, trim it until all supported browsers adjust. It sounds easy enough to do, just trim the value.

BTW, Firefox behavior is still buggy as it only trims leading spaces and leaves trailing spaces as they are, they have a bug for this: https://bugzilla.mozilla.org/show_bug.cgi?id=1728165.

Would you like to try to submit a fix by yourself?

**Comment by mgol** [comment]
BTW, [we have tests for whitespace handling of CSS Custom Properties](https://github.com/jquery/jquery/blob/175db73ec7938e774d9e93d3afdfb35a24466b47/test/unit/css.js#L1780) (a.k.a. CSS Variables) and the Firefox 91 change is why our Travis tests are failing: https://app.travis-ci.com/github/jquery/jquery/jobs/535647620

**Comment by timmywil** [comment]
I'm in favor of trimming if that's what all browsers will do eventually.

**Comment by fecore1** [comment]
@mgol Thanks for the quick response. 

> There was a recent spec change that require CSS Custom Properties values to be trimmed and it's just been implemented in Firefox 91 so far. 

Yes, so in my option, test case [css(--customProperty)](https://github.com/jquery/jquery/blob/main/test/unit/css.js#L1780) is  more appropriate to be modified  to `assert.equal( $elem.css( "--prop2" ), "val2", "Trim whitespace" );` because Travis tests has failed now and all browsers will implement the spec change eventually.

**Comment by mgol** [comment]
@fecore1 Yes, that sounds correct. Would you like to submit a PR with such a change?

**Comment by mgol** [comment]
@fecore1 Hey, since this breaks our CI, we'd rather have it fixed quickly. If you're not signing up to provide a PR, that's fine, but it'd be nice if you made it clear. So far I asked twice but you haven't replied.

**Comment by fecore1** [comment]
@mgol Sorry for the late reply because I'm on a business trip these days. I will try to provide a fix PR

**Comment by mgol** [comment]
@fecore1 Cool! Let me know if you need any help.

**Comment by mgol** [comment]
Link to that spec change issue: https://github.com/w3c/csswg-drafts/issues/774

**Comment by mgol** [comment]
Reopening until we discuss whether the currently landed solution is complete or if it needs more iterations and until we cherry-pick the change to `3.x-stable`. See https://github.com/jquery/jquery/pull/4938#issuecomment-928344401 for more context.

**Comment by mgol** [comment]
We settled on leaving the current solution as-is. We thought there are some issues with using the wrong CSS whitespace type but [it looks like it works just fine](https://github.com/jquery/jquery/pull/4956).

The PR landed on `main` in efadfe991a5c287af561a9326bf1427d726c91c1 and on `3.x-stable` in 219ccf5c5ffd751adb782335f0a36da79070f102 (with a followup hotfix in 509eeb892c85fca0f489a2af5bf334285e9330de).

Thanks, @fecore1!

**Comment by goarblt** [comment]
hello
https://www.evil.com
